### PR TITLE
Implement cli tool test mocker for LUIS

### PR DESCRIPTION
**Pull request by JuanAr**
## Proposed Changes
- Created cli test mocker tool
  - strip process args to get mock file
  - Leverage `nock` to intercept external service calls
  - Feed `nock` with json files with mock responses
  - Integrated specs test runner with test mocker
  - Integrated LUIS CLI test suite with test mocker
- Updated `mocha.opts` to select `**/*.suite.js` files only
- Added sample test mock files (more coming in new PRs to fully disconnect from external service)

**Note**: once tested and approved this mechanism can be used to test all cli tools disconnected

### CODE REVIEW

**Comment by justinwilaby** [comment]
Does this PR supersede #597 ?

**Comment by justinwilaby** [comment]
This PR has a failed build due to a malformed json file somewhere.

**Comment by JuanAr** [comment]
> Does this PR supersede #597 ?

Yes. But thinking again this, we might want to separate concerns: This PR is specifically to disconnect CLI unti tests from external services. While #597 is meant to simplify the way we create CLI unit tests.

**Comment by JuanAr** [comment]
> This PR has a failed build due to a malformed json file somewhere.

Fixed. The malformed files was LUIS/package-lock.json

File: packages/LUIS/test/luis.cli.specs.suite.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const path = require('path');
+const runTests = require('./testRunner.js').runTests;
```
[the remark this answers is not in the document]
Comment (JuanAr) [reply packages/LUIS/test/luis.cli.specs.suite.js]: We removed this file from this PR. However, we applied your feedback to #597 .

**Comment by cleemullins** [comment]
I don't know where this PR is at. It's ancient, so I'm closing it. 

@juanar - if we should bring this back, please let me know.

**Pull request closed without merging**